src/app/shared/classes/logout-base-page.ts

```diff
@@ -0,0 +1,37 @@
+import { BasePageComponent } from './base-page';
+import { AlertController, Platform } from '@ionic/angular';
+import { AuthenticationProvider } from '../../providers/authentication/authentication';
+import { Router } from '@angular/router';
+
+export abstract class LogoutBasePageComponent extends BasePageComponent {
```

Comment:
not sure what the purpose of this new abstract class is, or why it's needed

Reply:
its was how current app implements the logout functionality on any page that needed it.  Can be moved to a common module

Reply:
moved this into a provider